lib/ruby_lsp_rails/railtie.rb

```diff
@@ -2,15 +2,20 @@
 # frozen_string_literal: true
 
 require "rails/railtie"
-require "ruby_lsp_rails/middleware"
+require "ruby_lsp_rails/rack_app"
 
 module RubyLsp
   module Rails
     class Railtie < ::Rails::Railtie
-      initializer "ruby_lsp_rails.setup" do |app|
-        app.config.middleware.insert_after(ActionDispatch::ShowExceptions, RubyLsp::Rails::Middleware)
+      initializer "ruby_lsp_rails.setup" do |_app|
+        config.after_initialize do |app|
+          unless app.config.x.ruby_lsp_rails.server == false
+            app.routes.prepend do
+              T.bind(self, ActionDispatch::Routing::Mapper)
+              mount(RackApp.new => RackApp::BASE_PATH)
+            end
+          end
 
-        config.after_initialize do |_app|
           if defined?(::Rails::Server)
```

Comment:
Why do we need to do this?

Reply:
If we start the app with `bin/rails console` then the next line would fail with `uninitialized constant Rails::Server`.